.github/workflows/test.yml

```diff
@@ -148,53 +132,22 @@ jobs:
+              if: ${{ env.IS_FEATURE_BRANCH_PR == 'true' }}
               working-directory: ${{ matrix.tests.path }}
               run: dfx start --clean --background --host 127.0.0.1:8000 --artificial-delay 0
 
-            - name: Start dfx with artifical delay
-              if: ${{ env.IS_RELEASE_CANDIDATE }}
+            - name: Start dfx with artifcial delay
+              if: ${{ env.IS_RELEASE_BRANCH_PR == 'true' }}
               working-directory: ${{ matrix.tests.path }}
               run: dfx start --clean --background --host 127.0.0.1:8000
 
-            - name: Run Test
+            - name: Run test
               run: |
-                  BRANCH_NAME="${{ github.head_ref }}"
-                  MAIN_BRANCH='refs/heads/main'
-                  COMMIT_MESSAGE="${{ github.event.head_commit.message }}"
-                  RELEASE_TAG='release--'
-                  RELEASE_PR_MERGE_MSG='Merge pull request'
-                  RELEASE_REPO_MSG='demergent-labs/release--'
-
-                  IS_FEATURE_PR=true
-                  IS_RELEASE_PR=false
-                  IS_MERGE_TO_MAIN_FROM_RELEASE=false
-
-                  # Check if it's a feature pull request
-                  if [[ $BRANCH_NAME != *"$RELEASE_TAG"* && ${{ github.ref }} != "$MAIN_BRANCH" && !($COMMIT_MESSAGE == *"$RELEASE_PR_MERGE_MSG"* && $COMMIT_MESSAGE == *"$RELEASE_REPO_MSG"*) ]]; then
-                      IS_FEATURE_PR=true
-                  else
-                      IS_FEATURE_PR=false
-                  fi
-
-                  # Check if it's a release candidate pull request
-                  if [[ $BRANCH_NAME == *"$RELEASE_TAG"* && ${{ github.ref }} != "$MAIN_BRANCH" && !($COMMIT_MESSAGE == *"$RELEASE_PR_MERGE_MSG"* && $COMMIT_MESSAGE == *"$RELEASE_REPO_MSG"*) ]]; then
-                      IS_RELEASE_PR=true
-                  else
-                      IS_RELEASE_PR=false
-                  fi
-
-                  # Check if it's a merge from a release candidate into main
-                  if [[ ${{ github.ref }} == "$MAIN_BRANCH" && $COMMIT_MESSAGE == *"$RELEASE_PR_MERGE_MSG"* && $COMMIT_MESSAGE == *"$RELEASE_REPO_MSG"* ]]; then
-                      IS_MERGE_TO_MAIN_FROM_RELEASE=true
-                  else
-                      IS_MERGE_TO_MAIN_FROM_RELEASE=false
-                  fi
+                  IS_MERGE_TO_MAIN_FROM_RELEASE=${{ env.IS_MAIN_BRANCH == 'true' && contains(github.event.head_commit.message, 'Merge pull request') && contains(github.event.head_commit.message, 'demergent-labs/release--') }}
 
-                  # Determine the number of runs based on the conditions
-                  if [[ $IS_MERGE_TO_MAIN_FROM_RELEASE == true ]]; then
+                  if $IS_MERGE_TO_MAIN_FROM_RELEASE; then
                       RUNS=100
-                  elif [[ $IS_RELEASE_PR == true ]]; then
+                  elif ${{ env.IS_RELEASE_BRANCH_PR == 'true' && env.IS_MAIN_BRANCH == 'false' }}; then
```

Comment:
Can you give a descriptive name to each and not use elif? Just use if for both cases. And set runs to 5 before anything else

Reply:
sure